aiida/backends/managers/access.py

```diff
@@ -0,0 +1,259 @@
+            if not filename.startswith(self._temp_prefix):
+                filepath = basepath / filename
+                process_data = ProcessData(filepath=filepath)
+                process_datadicts[process_data.pid] = process_data
+
+        if ignore_self and os.getpid() in process_datadicts:
+            process_datadicts.pop(os.getpid())
+
+        return process_datadicts
+
+    def get_locking_processes(self, ignore_self=True):
+        """Returns all processes attempting to lock the profile
+
+        :param ignore_self:
+            flag to avoid returning the process that is requesting the information in the output
+            (enabled by default).
+        """
+        return self.get_processes_recorded_in(self._path_to_locking, ignore_self=ignore_self)
+
+    def get_tracked_processes(self, ignore_self=True):
+        """Returns all processes recorded when trying to access the profile
+
+        :param ignore_self:
+            flag to avoid returning the process that is requesting the information in the output
+            (enabled by default).
+        """
+        return self.get_processes_recorded_in(self._path_to_tracked, ignore_self=ignore_self)
+
+    def acquire_lock(self):
+        """Creates a locking file for the running process"""
+        tracked_processes = self.get_tracked_processes()
+        filtered_processes = self.separate_processes(tracked_processes)
+        self.delete_process_records(filtered_processes['terminated'])
+
+        if len(filtered_processes['remaining']) > 0:
+            raise exceptions.LockingProfileError('can`t lock profile')
+
+        process_data = ProcessData()
+        process_idn = process_data.pid
+        filepath_temp = self._path_to_locking / f'request_{process_idn}.pid'
+        filepath_final = self._path_to_locking / f'{process_idn}.pid'
+        process_data.write_to_file(filepath_temp)
+        os.rename(filepath_temp, filepath_final)
+
+        locking_processes = self.get_locking_processes()
+        if len(locking_processes) > 0:
+            os.remove(filepath_final)
+            raise exceptions.LockedProfileError('can`t lock profile')
+
+        tracked_processes = self.get_tracked_processes()
+        if len(tracked_processes) > 0:
+            os.remove(filepath_final)
+            raise exceptions.LockedProfileError('can`t lock profile')
+
+    def release_lock(self):
+        """Releases the locking file of the running process"""
+        process_data = ProcessData()
+        process_idn = process_data.pid
+        filepath = self._path_to_locking / f'{process_idn}.pid'
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(filepath)
+
+    @contextlib.contextmanager
+    def profile_locking_context(self):
+        """Context for working with a locked profile"""
+        try:
+            self.acquire_lock()
+            yield
+        finally:
+            self.release_lock()
+
+    def record_process_access(self, verify_lock=True):
```

Comment:
Do we have cases in which we want to use `verify_lock=False`? Maybe it's OK to keep it internally, but I wouldn't even expose it to the "outside" of the class for now, until there's a good usecase for it?